sdk/nodejs/output.ts

```diff
@@ -178,10 +173,19 @@ class OutputImpl<T> implements OutputInstance<T> {
         allResources = allResources || Promise.resolve([]);
         const allResourcesCopy = allResources.then(r => utils.union(copyResources(r), resourcesCopy));
 
+        // We are only known if we are not explicitly unknown and the resolved value of the output
+        // contains no distinguished unknown values.
+        isKnown = Promise.all([isKnown, promise]).then(([known, val]) => known && !containsUnknowns(val));
+
+        const lifted = Promise.all([allResourcesCopy, promise, isKnown, isSecret])
```

Comment:
What is the goal of `lifted` - and why do we want to have every one of our `Promise<>` properties depend on all of these promises?  I can't think of any problem with it - but it's also a non-trivial change and I'm unclear exactly why we need it.

Reply:
I added this b/c I am creating `Output`s from `Promise`s and returning them directly. In the original code, we would do something like

```
return new Output(promise).apply(v => output(v));
```

Now we do this:

```
return new Output(promise);
```

In the first case, the `apply` would ensure that if the promise resolved to an output, the inner output would be lifted onto the outer output. In the second case, there is no `apply`, so the `Output` constructor itself must handle the lifting. Strictly speaking, the approach I have here is "more correct" than the original code, as outputs will now never resolve to other outputs. Which I now realize is a breaking change, and which may end up _increasing_ our allocaitons-per-output. So maybe the original approach is a better choice--thoughts?

Reply:
(@CyrusNajmabadi interested in your thoughts on this point as well)

Reply:
Yes.  this approach seems far better.   👍